torch/lib/c10d/ProcessGroup.cpp

```diff
@@ -33,13 +33,20 @@ std::vector<at::Tensor> ProcessGroup::Work::result() const {
 
 void ProcessGroup::Work::synchronize() {}
 
-void ProcessGroup::Work::wait() {
+bool ProcessGroup::Work::wait() {
   std::unique_lock<std::mutex> lock(mutex_);
-  cv_.wait(lock, [&] { return completed_; });
+  cv_.wait(lock, [&] { return aborted_ | completed_; });
   if (exception_) {
     std::rethrow_exception(exception_);
   }
   synchronize();
+  return !aborted_;
```

Comment:
Should we change these default implementations given that we're only aiming to make this modification specific to `ProcessGroupGloo`?

Reply:
Good point. We may not be able to test it exhaustedly. Let me just raise if a user calls the ProcessGroup::abort just like NCCL and MPI.